python/lsst/meas/algorithms/brightStarStamps.py

```diff
@@ -42,17 +39,24 @@ def __str__(self):
         return self.name
 
 
-class BrightStarStamp(NamedTuple):
+class BrightStarStamp(Stamp):
     """Single stamp centered on a bright star, normalized by its
     annularFlux.
     """
-    starStamp: afwImage.maskedImage.MaskedImageF
     gaiaGMag: float
     gaiaId: int
     annularFlux: float
 
+    @classmethod
+    def factory(cls, stamp, metadata, idx):
+        gaiaGMags = metadata.getArray("G_MAGS")
+        gaiaIds = metadata.getArray("GAIA_IDS")
+        annularFluxes = metadata.getArray("ANNULAR_FLUXES")
+        return cls(stamp=stamp, ra=Angle(numpy.nan), dec=Angle(numpy.nan), size=-1,
+                   gaiaGMag=gaiaGMags[idx], gaiaId=gaiaIds[idx], annularFlux=annularFluxes[idx])
+
 
-class BrightStarStamps(collections.abc.Sequence):
+class BrightStarStamps(Stamps):
```

Comment:
It's not clear that `Stamps` is giving you much except a generic metadata element (that I think may be better avoided) and `readFits()` and `writeFits()` methods.  You might be better off having this be explicitly based on a `list` of `BrightStarStamp` with additional attributes and then having external `readFits()` and `writeFits()` methods that can handle the list (and may need to be overridden anyway in this class).

Reply:
I'm specifically trying to not have the `readFits` and `writeFits` methods overridden in this class.